src/terrain/routes/tags.clj

```diff
@@ -10,32 +13,76 @@
 
-   (GET "/filesystem/entry/:entry-id/tags" [entry-id]
-     (tags/list-attached-tags entry-id))
+      (DELETE "/tags" []
+        :summary schema/DeleteTagsSummary
+        :description schema/DeleteTagsDescription
+        :return nil
+        (ok (tags/remove-all-attached-tags)))
 
-   (PATCH "/filesystem/entry/:entry-id/tags" [entry-id type :as {body :body}]
-     (tags/handle-patch-file-tags entry-id type body))
+      (GET "/:entry-id/tags" []
+        :path-params [entry-id :- TargetIdParam]
+        :summary schema/GetAttachedTagSummary
+        :description schema/GetAttachedTagDescription
+        :return schema/TagList
+        (ok (tags/list-attached-tags entry-id)))
 
-   (GET "/tags/suggestions" [contains limit]
-     (tags/suggest-tags contains limit))
+      (PATCH "/:entry-id/tags" []
+        :path-params [entry-id :- TargetIdParam]
+        :query [params schema/TagTypeEnum]
+        :body [body schema/TagIdList]
+        :summary schema/PatchTagsSummary
+        :description schema/PatchTagsDescription
+        :return nil
+        (ok (tags/handle-patch-file-tags entry-id params body))))
 
-   (GET "/tags/user" []
-     (tags/list-user-tags))
+   (context "/tags" []
+            :tags ["tags"]
+     (GET "/suggestions" []
+        :query [params (dissoc schema/TagSuggestQueryParams :user)]
+        :summary schema/GetTagSuggestionsSummary
+        :description schema/GetTagSuggestionsDescription
+        :return schema/TagList
+        (ok (tags/suggest-tags (:contains params) (:limit params))))
 
-   (DELETE "/tags/user" []
-     (tags/delete-all-user-tags))
+     (GET "/user" []
+        :summary schema/GetUserTagsSummary
+        :description schema/GetUserTagsDescription
+        :return schema/TagList
+        (ok (tags/list-user-tags)))
 
-   (POST "/tags/user" [:as {body :body}]
-     (tags/create-user-tag body))
+     (DELETE "/user" []
+        :summary schema/DeleteUserTagsSummary
+        :description schema/DeleteUserTagsDescription
+        :return nil
+        (ok (tags/delete-all-user-tags)))
 
-   (PATCH "/tags/user/:tag-id" [tag-id :as {body :body}]
-     (tags/update-user-tag tag-id body))
+     (POST "/user" []
+        :body [body schema/TagRequest]
+        :summary schema/PostTagSummary
+        :description schema/PostTagDescription
+        :return schema/TagId
+        (ok (tags/create-user-tag body)))
 
-   (DELETE "/tags/user/:tag-id" [tag-id]
-     (tags/delete-user-tag tag-id))))
+     (PATCH "/user/:tag-id" []
+        :path-params [tag-id :- schema/TagIdPathParam]
+        :body [body schema/TagUpdateRequest]
+        :summary schema/PatchTagSummary
+        :description schema/PatchTagDescription
+        :return nil
+        (ok (tags/update-user-tag tag-id body)))
+
+     (DELETE "/user/:tag-id" []
+        :path-params [tag-id :- schema/TagIdPathParam]
+        :summary schema/DeleteTagSummary
+        :description schema/DeleteTagDescription
+        :return nil
```

Comment:
The one issue I'm still having is these last 2 endpoints are supposed to return `nil`, but I'm actually getting back Elasticsearch responses.  Any suggestions on how to fix that would be appreciated.

Reply:
You can remove the `:return nil` bits and update the handlers to something like this:

``` clojure
    (tags/update-user-tag tag-id body)
    (ok)
```

Reply:
Oh! I didn't realize it'd be that easy 😊 Thanks!  I'll refactor